perf-tool/src/client.cpp

```diff
@@ -162,20 +187,67 @@ void Client::closeClient()
     close(socketFd);
     printf("Closed connection with server.\n");
 }
+void printUsage()
+{
+    printf(" --portNo<p>: Required Port number\n --log<l> Output file name\n --host<h> host\n --usage<u> show help\n");
+}
 
-int main(int argc, char const *argv[])
+int main(int argc, char *argv[])
 {
     string hostname = "localhost";
+    std::string clientLogPath = "clientLogs.json";
     int portno = ServerConstants::DEFAULT_PORT;
-
-    if (argc > 1)
-    {
-        portno = atoi(argv[1]);
-        if (argc > 2)
-            hostname = argv[2];
-    }
-
-    Client client(portno, hostname, true);
+    static struct option long_options[] = {
+        {"portNo", optional_argument, 0, 'p'},
+        {"host", optional_argument, 0, 'h'},
+        {"log", optional_argument, 0, 'l'},
+        {"usage", no_argument, 0, 'u'},
+        {0, 0, 0, 0}
+    };
+    int c, index;
+    while (true) {
+        c = getopt_long(argc, argv, ":hlpu", long_options, &index);
+        if (c == -1)
+            break;
+        switch (c) {
+            case 'h':
+                if (optarg)
+                    hostname = optarg;
+                else
+                    printf("No argument passed to hostname using default\n");
+                break;
+
+            case 'l':
+                if (optarg)
+                    clientLogPath = optarg;
+                else
+                    printf("No argument passed to log using default\n");
+                break;
+
+            case 'p':
+                if (optarg)
+                    portno = atoi(optarg);
```

Comment:
Please add curly brackets here. Our convention is that the `if` part and the `else` part should look the same.